File: iosApp/iosApp/BuildConfig.swift
import Foundation

/// Build configuration constants.
///
/// This provides access to build-time configuration values like API keys.
/// For native iOS, we'll read from a Configuration.plist file.
struct BuildConfig {
    static let shared = BuildConfig()
    
    private let configuration: [String: Any]
    
    private init() {
        // Try to load from Configuration.plist
        if let path = Bundle.main.path(forResource: "Configuration", ofType: "plist"),
           let dict = NSDictionary(contentsOfFile: path) as? [String: Any] {
            configuration = dict
        } else {
            // Fall back to Info.plist
            configuration = Bundle.main.infoDictionary ?? [:]
        }
    }
    
    var googleBooksApiKey: String {
        if let key = configuration["GOOGLE_BOOKS_API_KEY"] as? String, !key.isEmpty {
            return key
        }
        // Fall back to environment variable or empty string
        return ProcessInfo.processInfo.environment["GOOGLE_BOOKS_API_KEY"] ?? ""
    }
}

File: androidApp/src/main/java/com/browsy/config/BuildConfig.kt
package com.browsy.config

/**
 * Build configuration constants for Android app.
 *
 * This object provides access to build-time configuration values like API keys.
 */
object BuildConfig {
    /**
     * Google Books API key for authenticated requests.
     * This should be set via local.properties or environment variable.
     */
    val GOOGLE_BOOKS_API_KEY: String
        get() {
            // In production, this would be injected via BuildConfig from Gradle
            // For now, we'll read from system property or use empty string
            return System.getProperty("google.books.api.key") 
                ?: System.getenv("GOOGLE_BOOKS_API_KEY") 
                ?: ""
        }
}
